ocw/data_source/podaac.py

```diff
@@ -0,0 +1,113 @@
+    # NOTE: This assumes a 'standard' calendar. It's possible (likely?) that
+    # users will want to customize this in the future.
+    parsed_time = utime(units)
+    return [parsed_time.num2date(x) for x in time[:]]
+
+
+def load_dataset(variable, datasetId='', datasetShortName='', name=''):
+    '''Loads a Dataset from PODAAC
+
+    :param variable: The name of the variable to read from the dataset.
+    :type variable: :mod:`string`
+
+    :param datasetId: dataset persistent ID. datasetId or \
+        shortName is required for a granule search. Example: \
+        PODAAC-ASOP2-25X01
+    :type datasetId: :mod:`string`
+
+    :param shortName: the shorter name for a dataset. \
+        Either shortName or datasetId is required for a \
+        granule search. Example: ASCATA-L2-25km
+    :type shortName: :mod:`string`
+
+    :param name: (Optional) A name for the loaded dataset.
+    :type name: :mod:`string`
+
+    :returns: A :class:`dataset.Dataset` containing the dataset pointed to by
+        the OpenDAP URL.
+
+    :raises: ServerError
+    '''
+    # Downloading the dataset using podaac toolkit
+        podaac = Podaac()
+        path = os.path.dirname(os.path.abspath(__file__))
+        granuleName = podaac.extract_l4_granule(
+            datasetId=datasetId, shortName=datasetShortName, path=path)
+        path = path + '/' + granuleName
+        d = netcdf_dataset(path, mode='r')
+        dataset = d.variables[variable]
+
+    # By convention, but not by standard, if the dimensions exist, they will be in the order:
+    # time (t), altitude (z), latitude (y), longitude (x)
+    # but conventions aren't always followed and all dimensions aren't always present so
+    # see if we can make some educated deductions before defaulting to just pulling the first three
+    # columns.
+        temp_dimensions = map(lambda x: x.lower(), dataset.dimensions)
+        dataset_dimensions = dataset.dimensions
+        time = dataset_dimensions[temp_dimensions.index(
+            'time') if 'time' in temp_dimensions else 0]
+        lat = dataset_dimensions[temp_dimensions.index(
+            'lat') if 'lat' in temp_dimensions else 1]
+        lon = dataset_dimensions[temp_dimensions.index(
+            'lon') if 'lon' in temp_dimensions else 2]
+
+    # Time is given to us in some units since an epoch. We need to convert
+    # these values to datetime objects. Note that we use the main object's
+    # time object and not the dataset specific reference to it. We need to
+    # grab the 'units' from it and it fails on the dataset specific object.
+        times = np.array(convert_times_to_datetime(d[time]))
+        lats = np.array(d.variables[lat][:])
+        lons = np.array(d.variables[lon][:])
+        values = np.array(dataset[:])
+        origin = {
+            'source': 'PO.DAAC',
+            'url': 'podaac.jpl.nasa.gov/ws'
+        }
+
+    # Removing the downloaded temporary granule before creating the OCW
```

Comment:
@lewismc these comments are code specific implementation information which is generally abstracted from the user, this was just for improving the readability of the code. Thanks.